From ea39b9684806c4858c20080ee80ccbecda11686a Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho@docker.com>
Date: Tue, 14 Mar 2017 20:24:47 -0700
Subject: [PATCH] libselinux: use python instead of ln --relative

busybox' ln doesn't support --relative, so we need to compute this install
path some other way for use with Alpine Linux.

Signed-off-by: Tycho Andersen <tycho@docker.com>
---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/libselinux/src/Makefile
index a277b87..7884116 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -182,7 +182,7 @@ install: all
 	install -m 755 $(LIBSO) $(SHLIBDIR)
 	test -d $(LIBDIR)/pkgconfig || install -m 755 -d $(LIBDIR)/pkgconfig
 	install -m 644 $(LIBPC) $(LIBDIR)/pkgconfig
-	ln -sf --relative $(SHLIBDIR)/$(LIBSO) $(LIBDIR)/$(TARGET)
+	ln -sf $(shell $(PYTHON) -c "import os.path; print os.path.relpath('$(SHLIBDIR)', '$(LIBDIR)')")/$(LIBSO) $(LIBDIR)/$(TARGET)
 
 install-pywrap: pywrap
 	test -d $(PYSITEDIR)/selinux || install -m 755 -d $(PYSITEDIR)/selinux
-- 
2.11.1

